compiler/damlc/tests/daml-test-files/TemplateTypeRep.daml

```diff
@@ -0,0 +1,46 @@
+-- Copyright (c) 2019, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- @SINCE-LF 1.dev
+daml 1.2 module TemplateTypeRep where
+
+import DA.Action
+
+template T1
+  with
+    x : Int
+    p : Party
+  where
+    signatory p
+
+template T2
+  with
+    y : Text
+    p : Party
+  where
+    signatory p
+
+template Template t => GenericT t
+  with
+    wrapped : t
+    p : Party
+  where
+    signatory p
+
+template instance GT1 = GenericT T1
+template instance GT2 = GenericT T2
+
+main = scenario do
+  unless (templateTypeRep @T1 == templateTypeRep @T1) $
```

Comment:
You can simplify these tests using `templateTypeRep @T1 === templateTypeRep @T1`.

Reply:
Not without adding a `Show` instance to `TemplateTypeRep`. I'm not sure if we want that though.

Reply:
Oh I see, yeah I thought about whether we want `Show` or not but haven’t quite made up my mind. I think it’s probably fine to add it given that I wouldn’t expect most users to ever use `templateTypeRep` and for the “expert” users it might be useful for debugging.

Reply:
I've added the `Show` instance and changed the tests to use `===`, I've also added `=/=` for the inequality tests.